### PR TITLE
Styling and Modal

### PR DESCRIPTION
**Pull request by wooseok0717**
- Added Modal for asking questions
- modal can make axios request to the server.
- added some styling.
- pull out update questions and update answers out of to a helper function to pass it as props and use it else where.

### CODE REVIEW

File: client/src/components/QuestionsAnswers.jsx [hunk 1]
```diff
@@ -9,30 +10,36 @@ export default function QuestionsAnswers({ productID }) {
...
+      <h3>Questions & Answers</h3>
       <QuestionsSearch
         questionList={questionList}
         setFilteredList={setFilteredList}
       />
-      <QuestionList questionList={filteredList} />
-      <AskQuestion />
+      <QuestionList questionList={filteredList} updateQuestions={updateQuestions} />
+      <button type="button" onClick={() => { setShow(true); }}> Ask a question</button>
+      <AskQuestion show={show} setShow={setShow} productID={tempId} update={updateQuestions} />
```
Comment (quanjingchen) [inline client/src/components/QuestionsAnswers.jsx]: on line 42, how about using conditional statement for this AskQuestion modal? like {show && <AskQustion .../>

File: client/src/components/QuestionsAnswers.jsx [hunk 2]
```diff
@@ -9,30 +10,36 @@ export default function QuestionsAnswers({ productID }) {
...
       .catch((err) => {
         console.log(err);
       });
+  };
+  useEffect(() => {
+    updateQuestions();
   }, [tempId]);
 
   useEffect(() => {
     setFilteredList(questionList);
```
Comment (quanjingchen) [inline client/src/components/QuestionsAnswers.jsx:29]: on line 30, it seems that filteredList is the same as the questionList? i thought that there should be some filter function?

File: client/src/components/Sub_Questions/styles/questions.css [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+  left: 0;
+  top: 0;
+  right: 0;
+  bottom: 0;
+  background-color: rgba(0, 0, 0, 0.5);
+  display: flex;
+  align-items: center;
+  justify-content: center;
+}
+
```
Comment (quanjingchen) [inline client/src/components/Sub_Questions/styles/questions.css:30]: I'm gonna steal this

File: client/src/components/QuestionsAnswers.jsx (hunk 2 above)
Comment (wooseok0717) [reply client/src/components/QuestionsAnswers.jsx:29]: I don't know if I understood your question correctly but this is just presetting my filtered list and later on it's state will be changed by other functions
and my list will stay the same for returning back to original state.

Comment (quanjingchen) [reply client/src/components/QuestionsAnswers.jsx:29]: gotcha

File: client/src/components/Sub_Questions/styles/questions.css (hunk 3 above)
Comment (wooseok0717) [reply client/src/components/Sub_Questions/styles/questions.css:30]: <3

File: client/src/components/QuestionsAnswers.jsx (hunk 1 above)
Comment (wooseok0717) [reply client/src/components/QuestionsAnswers.jsx]: done 👍

**Review by quanjingchen — APPROVED** [review]

**Pull request merged**